Documentation (#28)
NSF-Swift/satellite-overhead: README.md

Reviewer: Add a section on how to pull TLE files manually?

# S.O.P.P. - Satellite Orbit Prediction Processor

SOPP is an open-source tool for calculating satellite interference to radio astronomy observations.

![alt text](https://github.com/NSF-Swift/satellite-overhead/blob/documentation/FBD.png?raw=true)

## Running SOPP

### Install the requirements:
```bash
>> pip3 install -r requirements.txt
```

### Setting up the supplemental files

#### Active Satellites TLE File
There should be a [TLE](https://en.wikipedia.org/wiki/Two-line_element_set) file, containing the satellites that you 

> want to search through, at `supplements/satellites.tle` under the root directory. If you do not specify a config
> file, SOPP will download a list of TLEs from Space-trak.org

Reviewer: ```suggestion
want to search through, at `supplements/satellites.tle` under the root directory. If you do not specify a config
file, SOPP will download a list of TLEs from celestrak.org.
```

Author: Changed

#### Config file
There should be a file at `supplements/.config` under the root directory to set default observation values for a reservation. 
The following is an example of a config file:

>     [RESERVATION]

Reviewer: Can @ellem95 add descriptions for each of these configuration keys?

Author: Added!

- **Celestrak:** [Celestrak](celestrak.org) is the precursor to space-track and is available without an account. While it has been predicted to be phased out for years, it is still actively maintained. It also has an easily downloadable list of active satellites, which doesn’t include space debris.
- **Astropy:** [Astropy](https://www.astropy.org/) is a Python package designed for use in astronomy and contains functions for many space math problems. Astropy is used in the SOPP tool to calculate the path of an observation target through the sky to determine the telescope’s movement as a function of time. These azimuth and elevation values are then used to determine if a satellite’s movement intersects with this path.
- **Rhodesmill Skyfield:** The [Skyfield API](https://rhodesmill.org/skyfield/) is a comprehensive Python package for computing the positions of stars, planets, and satellites. SOPP uses the Skyfield API to find all the satellites visible above the horizon during an observation search window.

> - **Auxiliary Frequency Database:** a satellite frequency database was created by scraping frequency information from various open sources. The code used to generated the database can be found [here](https://github.com/NSF-Swift/sat-frequency-scraper).

Reviewer: ```suggestion
- **Satellite Frequencies:** satellite frequencies were pulled by scraping frequency information from various open sources. The code used to generated the database can be found [here](https://github.com/NSF-Swift/sat-frequency-scraper).
```

Author: Changed

> ### Custom Data Classes

Reviewer: Seems like these should be [docstrings](https://peps.python.org/pep-0257/).

+ **Reservation:** stores the Facility, as well as some additional reservation-specific information, such as reservation start and end times.
  + *facility*: Facility (see above)
  + *time*: TimeWindow that contains the start and end time of the ideal reservation. Type TimeWindow (see above).
  + *frequency*: frequency of the requested observation. Type FrequencyRange.
+ **OverheadWindow:** designed to store the time windows that a given satellite is overhead and includes the Satellite object, as well as a TimeWindow object that contains the interference start and end times.
  + *satellite*: the satellite that is overhead during the time window. Type Satellite.
  + *overhead_time*: the time the satellite enters and exits view. TimeWindow.
+ **ObservationPath:** represents the telescope's position as a function of time by storing an altitude, azimuth, and timestamp
  + *altitude*: altitude of the telescope. Angle (type from Astropy).
  + *azimuth*: azimuth of the telescope. Angle (from Astropy).
  + *time*: time at which the telescope is at the stored altitude and azimuth. datetime.

> ### Functional Modules

Reviewer: Seems like these should be [docstrings](https://peps.python.org/pep-0257/).

Author: Moved all of these into their respective files and deleted from README
